test/Microsoft.AspNetCore.Mvc.Razor.Extensions.Test/TestCompilation.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.Extensions.DependencyModel;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Extensions
+{
+    public static class TestCompilation
```

Comment:
We get the metadata references by getting the current assembly and doing `DependencyContext.Load()`. Not sure if I can do that from outside the assembly. I can give this a shot.